src/mpid/ch4/shm/xpmem/xpmem_send.h

```diff
@@ -12,10 +12,11 @@
 #include "xpmem_pre.h"
 #include "xpmem_impl.h"
 
-MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_lmt_isend(const void *buf, MPI_Aint count,
-                                                   MPI_Datatype datatype, int rank, int tag,
-                                                   MPIR_Comm * comm, int context_offset,
-                                                   MPIDI_av_entry_t * addr, MPIR_Request ** request)
+MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_lmt_coop_isend(const void *buf, MPI_Aint count,
+                                                        MPI_Datatype datatype, int rank, int tag,
+                                                        MPIR_Comm * comm, int context_offset,
+                                                        MPIDI_av_entry_t * addr, int packet_type,
+                                                        MPIR_Request ** request)
 {
```

Comment:
Wesley suggested me to use `MPIDI_XPMEM_lmt_isend` in shm layer and let it call `MPIDI_XPMEM_lmt_coop_isend` in XPMEM layer so that the design is clear and consistent. 
If we change back to `MPIDI_XPMEM_lmt_isend`, I still need to change its API (add `packet_type` input parameter) if you think it's ok to do it.

Reply:
I can look at the logic again once you cleaned up the PR. If you want to rename the function name, you need to put it into a separate commit.